lib/browser/api/menu-item.js

```diff
@@ -7,18 +7,13 @@ let nextCommandId = 0
 const MenuItem = function (options) {
   const {Menu} = require('electron')
 
-  this.selector = options.selector
-  this.type = options.type
-  this.role = options.role
-  this.label = options.label
-  this.sublabel = options.sublabel
-  this.accelerator = options.accelerator
-  this.icon = options.icon
-  this.enabled = options.enabled
-  this.visible = options.visible
-  this.checked = options.checked
-
-  this.submenu = options.submenu
+  // Clone and sanitize the provided options
+  options = Object.assign({}, options)
+  delete options.overrideProperty
+  delete options.overrideReadOnlyProperty
```

Comment:
I tried that technique, but because `hasOwnProperty` only checks the object itself, and doesn't walk up the prototype chain, it was still possible to override methods like `overrideProperty`. I just updated it to use `!(key in this)` as a check instead.